### ISSUE TITLE
Enum Or Set value save which contian & character

### ISSUE THREAD

**Issue by armenmargaryan** [phpmyadmin/phpmyadmin#16977]
### Describe the bug

If enum or set field contain & character it is not possible to save such value from table linar editing.
i have tested this beaver on latest phpmyadmin version

### To Reproduce

1. create such table with enum or set field type and set value options to be test&test and test2&test
2. insert few record to test table with values test&test 
3. open this table list page and double click on row enum field and try to choose second option test2&test you will see that instead your chosen **test2&test** there going to be saved **test2& amp;test**
4. ERROR: after trying to save record you will get error or value will be not saved

### Expected behavior

Values must be saved

### Screenshots

Here is screenshot of save error

![issue](https://user-images.githubusercontent.com/6891642/122886746-46e29c00-d351-11eb-87a1-0d0c8363735e.png)

### Server configuration

- Operating system: Windows, Linux
- Web server: apache
- Database version: tested on mysql 5.7,8.0
- PHP version: 7.2,7.4
- phpMyAdmin version: 5.1.0

### Client configuration

- Browser: Firefox and Crome
- Operating system: Windows

**Comment by williamdes** [comment]
Thank you for the report !
Confirmed with
```sql
START TRANSACTION;
INSERT INTO `enums` (`enum`, `set`, `ser`) VALUES
('', 'b&c', 1),
('', 'b&c', 2),
('a&b', 'a&b,b&c', 3);

ALTER TABLE `enums`
  ADD UNIQUE KEY `ser` (`ser`);

ALTER TABLE `enums`
  MODIFY `ser` bigint(20) UNSIGNED NOT NULL AUTO_INCREMENT, AUTO_INCREMENT=4;
COMMIT;
```

**Comment by williamdes** [comment]
Hi @armenmargaryan 
I generated the snapshot, the fix is included in it: [(phpMyAdmin 5.1+snapshot)](https://www.phpmyadmin.net/downloads/#snapshot_5.1+snapshot)

Let me know if it works fine

**Comment by armenmargaryan** [comment]
Hi @williamdes ,
Yes i can confirm that with new build its resolved!
Thank you for so fast response and fix.
